nextgen/task/src/task.rs

```diff
@@ -159,6 +166,43 @@ impl Task {
         Ok(false)
     }
 
+    /// Return a list of all project-relative input files for
+    /// the task
+    pub fn get_input_files<S: AsRef<str>>(
+        &self,
+        workspace_root: &Path,
+        project_source: S,
+    ) -> miette::Result<Vec<ProjectRelativePathBuf>> {
+        let mut list = vec![];
+        let project_source = project_source.as_ref();
+
+        for path in &self.input_files {
+            list.push(path.strip_prefix(project_source).unwrap().to_owned());
```

Comment:
I think this should return the paths relative from the workspace root.

The "relative from the project root" is only required for the affected logic, so I would move that part over there.